libs/portals/admin/ids-admin/src/components/Client/Permissions.tsx

```diff
@@ -138,7 +131,21 @@ function Permissions({ data = mockData }: PermissionsProps) {
           </T.Table>
         </ShadowBox>
       )}
-      <AddPermissions onClose={handleModalClose} isVisible={isModalVisible} />
+      <input
```

Comment:
A better pattern handling hidden input arrays is the following way.

```
          {addedScopes.map(({ name }) => (
              <input
                key={env}
                type="hidden"
                name="addedScopes"
                value={name}
              />
            ))}
```

Then in the action

const schema = {
    ...,
    addedScopes: zfd.repeatable(z.optional(z.array(z.string())),
} 
```

That way you don't have to do the transformations back and forth

Reply:
That is awesome, thank you for that tip!